jvm.c

```diff
@@ -783,10 +786,122 @@ stack_entry_t *execute(method_t *method,
+            uint16_t index = ((param1 << 8) | param2);
+
+            char *field_name, *field_descriptor, *class_name;
+            class_name = find_field_info_from_index(index, clazz, &field_name,
+                                                    &field_descriptor);
+
+            /* skip java.lang.System in order to support java print
+             * method */
+            if (!strcmp(class_name, "java/lang/System")) {
+                pc += 3;
+                break;
+            }
+
+            class_file_t *target_class =
+                find_or_add_class_to_heap(class_name, prefix);
+            field_t *field =
+                find_field(field_name, field_descriptor, target_class);
+
+            switch (field_descriptor[0]) {
+            case 'B':
+                /* signed byte */
+                push_byte(op_stack, field->variable->value.char_value);
+                break;
+            case 'C':
+                /* unicode character code */
+                push_byte(op_stack, field->variable->value.char_value);
+                break;
+            case 'I':
+                /* integer */
+                push_int(op_stack, field->variable->value.int_value);
+                break;
+            case 'J':
+                /* long integer */
+                push_long(op_stack, field->variable->value.long_value);
+                break;
+            case 'S':
+                /* signed short */
+                push_short(op_stack, field->variable->value.short_value);
+                break;
+            case 'Z':
+                /* true or false */
+                push_byte(op_stack, field->variable->value.char_value);
+                break;
+            case 'L':
+                /* an instance of class */
+                push_ref(op_stack, field->variable->value.ptr_value);
+                break;
+            default:
+                fprintf(stderr, "Unknown field descriptor %c\n",
+                        field_descriptor[0]);
+                exit(1);
+            }
             pc += 3;
             break;
+        }
+
+        /* Put static field to class */
+        case i_putstatic: {
+            uint8_t param1 = code_buf[pc + 1], param2 = code_buf[pc + 2];
+            uint16_t index = ((param1 << 8) | param2);
+
+            char *field_name, *field_descriptor, *class_name;
+            class_name = find_field_info_from_index(index, clazz, &field_name,
+                                                    &field_descriptor);
+
+            class_file_t *target_class =
+                find_or_add_class_to_heap(class_name, prefix);
+            field_t *field =
+                find_field(field_name, field_descriptor, target_class);
+
+            switch (field_descriptor[0]) {
+            case 'B':
+                /* signed byte */
+                field->variable->value.char_value = (u1) pop_int(op_stack);
+                field->variable->type = VAR_BYTE;
+                break;
+            case 'C':
```

Comment:
It seems that the Unicode character handling is incomplete. It should be distinguished from generic byte manipulation.

Reply:
Ok, I will change to 2 bytes to represent Unicode character.

Reply:
Add some comments starting with FIXME/TODO to address complete Unicode handling.